yjit/src/asm/mod.rs

```diff
@@ -69,20 +78,43 @@ pub struct CodeBlock {
 
 impl CodeBlock {
     /// Make a new CodeBlock
-    pub fn new(mem_block: VirtualMem, outlined: bool) -> Self {
-        Self {
-            mem_size: mem_block.virtual_region_size(),
+    pub fn new(mem_block: Rc<RefCell<VirtualMem>>, page_size: usize, outlined: bool) -> Self {
+        let mem_size = mem_block.borrow().virtual_region_size();
+        let mut cb = Self {
             mem_block,
+            mem_size,
             write_pos: 0,
+            page_size,
+            page_fault: false,
             label_addrs: Vec::new(),
             label_names: Vec::new(),
             label_refs: Vec::new(),
             #[cfg(feature = "asm_comments")]
             asm_comments: BTreeMap::new(),
-            #[cfg(feature = "disasm")]
             outlined,
             dropped_bytes: false,
```

Comment:
Question: are `page_fault` and `dropped_bytes` essentially the same thing?

Reply:
Similar but not exactly the same. `dropped_bytes` is used for (1) out of VirtualMem and (2) too large jump offset. Using the same flag for `page_fault` (inside `VirtualMem`, but out of the current code page) and (1) is probably acceptable, I'm not sure if we should share the same flag as (2). Perhaps we could just assert it and remove (2): `cb.dropped_bytes = true` in `write_jcc_ptr`.

Reply:
I agree it's very strange that we're using dropped_bytes when the offset doesn't fit in `write_jcc_ptr`. I think that should be an assertion 👍